### ISSUE TITLE
Tools automatic dependabot updates require validation of manifests

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#2643]
After #2600 and #2636 have been merged dependabot will create dependency updates for tools defined in [third_part/go.mod](https://github.com/Kong/kubernetes-ingress-controller/blob/main/third_party/go.mod).

The problem is that changes to versions of those tools might entail changes in manifests e.g. [all-in-one manifests](https://github.com/Kong/kubernetes-ingress-controller/tree/main/deploy/single).

In order to prevent diverging tools' versions and manifests (created by those tools) let's implement a mechanism that will automatically trigger manifests update in the same PR that the tool version bump is being done.

There is a [mechanism in place](https://github.com/Kong/kubernetes-ingress-controller/blob/03fe8c9bd6b313eefd2e46382065890e24a626b4/.github/workflows/test.yaml#L43-L47) already that will prevent the PRs that update either the tools or the manifests that would result in a diff when manifests are recreated so this types of PRs shouldn't get merged into `main`.
